oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elasticsearch/util/TermQueryBuilderFactory.java

```diff
@@ -67,11 +67,11 @@ private static TermQueryBuilder newFulltextQuery(String ft, String field) {
     }
 
     public static PrefixQueryBuilder newPrefixQuery(String field, @NotNull String value) {
-        return prefixQuery(keywordFieldName(field), value);
+        return prefixQuery(field, value);
```

Comment:
Since we are always storing the non-analyzed text, shouldn't we actually use `indexDefinition.getElasticKeyword` here so that prefix search could be made on appropriate field in every case (since prefix search doesn't work with analyzed text) ?

Reply:
You are right. But this should be done by the caller and not in this utility method. As you can see from the commit message, the refactoring on this has been done partially. It will be completed in the task to refactor the query code.